npc/re/instances/GhostPalace.txt

```diff
@@ -0,0 +1,982 @@
+	enablenpc instance_npcname("#sv");
+	enablenpc instance_npcname("#tv");
+	initnpctimer;
+	end;
+OnTimer3000:
+	npctalk "Voice of Princess Tiara : What can I do for you?";
+	end;
+OnTimer8000:
+	npctalk "Voice of Sakray : I will always stand by you. Please rest beside me.", instance_npcname("#sv");
+	end;
+OnTimer13000:
+	npctalk "Thanatos Magic Trace: Sakray, do not forget the covenant of blood between us.", instance_npcname("#tv");
+	end;
+OnTimer17000:
+	npctalk "Thanatos Magic Trace: Your soul is mine until you find the blood that satisfies me...", instance_npcname("#tv");
+	end;
+OnTimer22000:
+	npctalk "Voice of Sakray : Rest in the shadow with me.", instance_npcname("#sv");
+	end;
+OnTimer25000:
+	npctalk "Voice of Sakray : In the dark forever...", instance_npcname("#sv");
+	end;
+OnTimer28500:
+	disablenpc instance_npcname("#tb");
+	disablenpc instance_npcname("#sv");
+	disablenpc instance_npcname("#tv");
+	mapannounce 'map_spa$, "It is time to leave the Ghost Palace.", bc_map,0xFFFF00,FW_NORMAL,12;
+	enablenpc instance_npcname("Way out of Palace");
+	enablenpc instance_npcname("King#dk11");
+	end;
+}
+
+1@spa,195,41,1	duplicate(Lurid Royal Guard#dk)	Thanatos Magic Trace#dk3	CLEAR_NPC
+1@spa,198,43,1	duplicate(Lurid Royal Guard#dk)	#sv	CLEAR_NPC
+1@spa,196,41,1	duplicate(Lurid Royal Guard#dk)	#tv	CLEAR_NPC
+
+1@spa,204,29,1	script	King#dk11	4_M_RUSKING,{
+	disable_items;
+	mes "[King]";
+	mes "Hey there, are you alive?";
+	mes "I did not expect to meet someone who is still alive.";
+	next;
+	mes "[King]";
+	mes "Do you possibly have the ^006400Gray Shard^000000?";
+	mes "Can you give it to me?";
+	mes "The memories of the faithful royal guard";
+	mes "and the memories of everyone he loved";
+	mes "are contained inside of these shards.";
+	next;
+	mes "[King]";
+	mes "Will you give the ^006400Gray Shard^000000 to me?";
+	mes "The guilt is eating my soul.";
+	mes "My soul is stuck within the Ghost Palace.";
+	next;
+	mes "[King]";
+	mes "I am collecting Sakray's abandoned memories from these shards.";
+	next;
+	mes "[King]";
+	mes "If you give any ^006400Gray Shard^000000s to me, I can make equipment for you.";
+	next;
+	mes "[King]";
+	mes "The cursed knight...";
+	mes "Please help me save the soul of Sakray.";
+	next;
+	setarray .@item_name$[0],
+		"Thanatos Sword", "Thanatos Great Sword", "Thanatos Spear", "Thanatos Long Spear", "Thanatos Staff",
+		"Thanatos Two-handed Staff", "Thanatos Bow", "Thanatos Dagger", "Thanatos Katar", "Thanatos Knuckle",
+		"Thanatos Hammer", "Thanatos Axe", "Thanatos Violin", "Thanatos Whip",
+		"Gray Shield", "Gray Armor", "Gray Robe", "Gray Cloak", "Gray Shoes", "Gray Helmet";
+	setarray .@items_list[0],13441,21009,1438,1496,1669,2023,18119,13093,28000,1836,16028,28100,1933,1988,2187,15090,15091,20721,22033,18820;
+
+	for ( .@i = 0; .@i < 20; .@i++ )
+		.@menu$ += .@item_name$[.@i] + ":";
+	.@menu$ += "Cancel";
+
+	while(true) {
+		.@s = select(.@menu$) - 1;
+		if (.@s == 20) {
```

Comment:
`.@s == 20` -> `.@s < 0 || .@s >= 20` if anything goes wrong. Maybe It'd be better to replace the hardcoded `20` with something like `.@item_cnt`